### PR TITLE
Creating tenderTable

### PR DESCRIPTION
**Pull request by ABabayan32**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
-    <meta charset="UTF-8">
+      <link href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" rel="stylesheet">
```
Comment (lavrovpy) [inline index.html:4]: do we really need this link?

File: src/styles/main.css [hunk 2]
```diff
@@ -1,14 +1,29 @@
 
 .tenderFlexIcon {
-  background: url('@/assets/icons/TenderFlex -- Logo.png');
+  background: url('@/assets/icons/logo.png');
   display: block;
-  background-size: 300px;
-  width: 300px;
-  height: 59px;
-  border: 1px solid white;
+  background-size: 18.75rem;
```
Comment (lavrovpy) [inline src/styles/main.css:5]: hardcoding sizes is not a good way. better to set the parent component size and apply _height: 100%_ to a child component

File: src/styles/main.css [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+  width: 18.75rem;
```
Comment (lavrovpy) [inline src/styles/main.css:6]: also check out about min-width and max-width. They are much more flexible that width

File: src/styles/main.css [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+  height: 3.6875rem;
+  border: 0.0625rem solid white;
```
Comment (lavrovpy) [inline src/styles/main.css:8]: usually color names (like white) we use only for prototyping. ii is always better to use hex code instead

File: src/styles/main.css [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+  border-radius: 0.9375rem;
+  border: 0.0625rem solid #ccc;
+}
+.button {
+  display: flex;
+  background-color: #27aae1;
+  border: none;
+  font-size: 1.25rem;
+  cursor: pointer;
+  width: 12.5rem;
```
Comment (lavrovpy) [inline src/styles/main.css:21]: do not hardcode the width. make the button width depended on the inner text

File: src/views/ContractorHome/ContractorHome.vue [hunk 6]
```diff
@@ -1,32 +1,28 @@
...
+import Header from "@/views/Header/Header.vue";
+import TenderTable from "@/views/TenderTable/TenderTable.vue";
 </script>
-
 <script>
 export default {
   data() {
     return {
-      tendersShown: true,
+      tendersShown: true
```
Comment (lavrovpy) [inline src/views/ContractorHome/ContractorHome.vue:22]: tendersShown should be a computed property

File: src/views/Header/Header.vue [hunk 7]
```diff
@@ -11,8 +12,8 @@ export default {
       offerCount: 0,
     }
   },
-  async beforeCreate() {
-    this.tenderCount = await fetch('http://localhost:8080/tenders/me/count',{
+  async beforeMount() {
+    this.tenderCount = await fetch(API_BASE_URL + '/tenders/me/count',{
```
Comment (lavrovpy) [inline src/views/Header/Header.vue:16]: nice!

Comment (lavrovpy) [inline src/views/Header/Header.vue:16]: add try ... catch to handle errors here

File: src/views/Header/Header.vue [hunk 8]
```diff
@@ -74,21 +74,29 @@ export default {
...
-    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
   </header>
   <div class="header">
     <div class="iconContainer">
       <div class='tenderFlexIcon headerIcon'></div>
     </div>
+
     <div class="tendersButtonContainer">
-      <button @click="onTendersClick" class="tendersButton"><i style="color:white;" class="fa fa-heartbeat"></i> Tenders <span class="tenderCount">{{tenderCount}}</span></button> </div>
+      <button @click="onTendersClick" class="tendersButton"><button class="tendersButton"></button>
```
Comment (lavrovpy) [inline src/views/Header/Header.vue:84]: use camel case for js code and kebab case for css

File: src/views/Header/Header.vue [hunk 9]
```diff
@@ -74,21 +74,29 @@ export default {
...
+        <v-icon icon="fa fa-heartbeat" />
+        Tenders <span class="tenderCount">{{tenderCount}}</span></button> </div>
+
     <div class="offersButtonContainer">
-      <button @click="onOffersClick" class="offersButton"><i class="fa fa-commenting-o"></i> Offers<span class="offerCount">{{offerCount}}</span></button> </div>
+      <button @click="onOffersClick" class="offersButton"><button class="offersButton"></button>
```
Comment (lavrovpy) [inline src/views/Header/Header.vue:89]: onOffersClick is a bad method name

File: src/views/Header/Header.vue [hunk 10]
```diff
@@ -98,7 +106,7 @@ export default {
 </template>
 
 <style>
-@import "@/styles/header.css";
+@import "@/views/Header/header.css";
```
Comment (lavrovpy) [inline src/views/Header/Header.vue:109]: css should be inside Header.vue, not a separate file

File: src/views/Header/header.css [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    margin-top: -0.625rem;
+}
+
+.exitButton {
+    display: flex;
+    background-color: #27aae1;
+    border: none;
+    font-size: 1.75rem;
+    cursor: pointer;
+    width: 2.5rem;
```
Comment (lavrovpy) [inline src/views/Header/header.css:120]: avoid using hardcoded width and height for buttons

File: src/views/Login/Login.vue [hunk 12]
```diff
@@ -62,5 +62,5 @@ export default {
 
 <style scoped>
 @import "@/styles/main.css";
-@import "@/styles/login.css";
+@import "@/views/Login/login.css";
```
Comment (lavrovpy) [inline src/views/Login/Login.vue:65]: same here. styles specific to the login page should be inside Login.vue. If there are styles that are used in multiple components they should be global

File: src/views/TenderTable/TenderTable.vue [hunk 13]
```diff
@@ -0,0 +1,188 @@
+<template>
+  <div class="tableHeaderContainer"></div>
+  <div class="tenderTable">
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:3]: use kebab case

File: src/views/TenderTable/TenderTable.vue [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+        </th>
+        <th>
+          Offers <i class="bi bi-sort-alpha-down" aria-label='Sort Icon'></i>
+        </th>
+      </tr>
+      </thead>
+      <tbody>
+
+      <tr  v-for="tender in tenders" :key='tender'>
+        <td style="width: 30%;text-align: center; color: black">{{tender['awardFileKey']}}</td>
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:31]: better to create css classes and move styles. by the way I think there is a better way to set column width

File: src/views/TenderTable/TenderTable.vue [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+        <td style="width: 20%;text-align: center; color: #42474d">{{tender['name']}}</td>
+        <td style="width: 20%;text-align: center; color: #42474d">{{tender['tenderStatus'] ? tender['tenderStatus'].name : ''}}</td>
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:33]: better to use **dot notation** like tender.tenderStatus.name

File: src/views/TenderTable/TenderTable.vue [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+        <td style="width: 20%;text-align: center; color: #42474d">{{tender['deadForSinging'] ? formatDate(tender['deadForSinging']) : ''}}</td>
+        <td style="width: 10%;text-align: center; color: #42474d"><button @click="onTenderOffersClick(tender)"
+                                                                          class="offerButton rcorners1" type="button" >
+          {{responseDate(tender, offers)}}</button></td>
+      </tr>
+      </tbody>
+    </table>
+    <v-pagination
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:41]: nice!

File: src/views/TenderTable/TenderTable.vue [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+        :disabled = "tenders.length === 0"
+        v-model="page"
+        :length="(tenderCount%count !== 0) ? tenderCount/count+1 : tenderCount/count"
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:44]: move it to a computed property

File: src/views/TenderTable/TenderTable.vue [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+    }).then((response) => {
+          if (response.ok) {
+            return response.json()
+          } else if(response.status === 401 || response.status === 403){
+            logout();
+
+          }
+        }
+    )
+        .then(response => {
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:88]: add a catch block to handle errors

File: src/views/TenderTable/TenderTable.vue [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+          });
+    },
+    formatDate(miliseconds) {
+      let d = new Date(miliseconds);
+      return ("0" + d.getDate()).slice(-2) + "-" + ("0" + (d.getMonth() + 1)).slice(-2) + "-" +
+          d.getFullYear() + " " + ("0" + d.getHours()).slice(-2) + ":" + ("0" + d.getMinutes()).slice(-2);
+
+    },
+    responseDate(tender, offers) {
+      let id = +tender['id'];
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:174]: if you want to convert a string to a number use Number()

File: src/views/TenderTable/TenderTable.vue [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+              logout();
+            }
+          }
+      )
+          .then(response => {
+            return response;
+          });
+    },
+    formatDate(miliseconds) {
+      let d = new Date(miliseconds);
```
Comment (lavrovpy) [inline src/views/TenderTable/TenderTable.vue:168]: better to use smth like this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

**Review by lavrovpy — CHANGES_REQUESTED** [review]
good!

**Comment by ABabayan32** [comment]
I will merge requested changes with next commit.

**Pull request merged**